pkg/controller/miganalytic/miganalytics_controller.go

```diff
@@ -174,6 +176,8 @@ func (r *ReconcileMigAnalytic) Reconcile(request reconcile.Request) (reconcile.R
 		!analytic.Status.HasBlockerCondition(),
 		"The analytic is ready.")
 
+	analytic.Spec.Refresh = false
```

Comment:
Maybe I have missed the conversation, but I am wondering why do we need this spec field? Is there a way we can refresh the mic-analytics CR at a particular interval even without the refresh?

Reply:
@alaypatel07 there is no need to keep refreshing at particular interval. Before running a migration, we can never be sure whether the data currently shown in MigAnalytic is latest or not. We need a way to refresh MigAnalytic just-in-time.

Reply:
@pranavgaikwad if that is the case can we make the migmigration create the miganalytic CR? I do not like the  pattern of creating the `refresh` field unless absolutely needed on our CRs

Reply:
@alaypatel07 MigPlan will create MigAnalytic CR

Reply:
We definitely need a way to trigger analytics gathering on demand and only when needed in the PV migration configuration process. I think this accomplishes that. 

This explicit refresh trigger seems to give most precise control over when the trigger is sent. I guess we could always do the UI process of delete/create a new CR if that seems preferable, but this seems like less work for the APIserver.

One concern that comes to mind: we need to watch out for problems where the general-purpose MigAnalytics created by the UI today blocks the MigAnalytics controller reconcile for a while, and our "mini" MigAnalytics for PV size analysis is stuck in the queue behind it. This could cause the MigPlan creation wizard to spin for a while.